email.py

```diff
@@ -0,0 +1,265 @@
+	"""
+	logging.debug("Sending deferred email: %s" % str(emailKey))
+	queueEntity = db.Get(emailKey)
+	assert queueEntity, "Email queue object went missing!"
+	if queueEntity["isSend"]:
+		return True
+	elif queueEntity["errorCount"] > 3:
+		raise ChildProcessError("Error-Count exceeded")
+	transportFunction = conf["viur.email.transportFunction"] # First, ensure we're able to send email at all
+	assert callable(transportFunction), "No or invalid email transportfunction specified!"
+	try:
+		resultData = transportFunction(dests=queueEntity["dests"],
+									   sender=queueEntity["sender"],
+									   cc=queueEntity["cc"],
+									   bcc=queueEntity["bcc"],
+									   subject=queueEntity["subject"],
+									   body=queueEntity["body"],
+									   headers=queueEntity["headers"],
+									   attachments=queueEntity["attachments"])
+	except Exception:
+		# Increase the errorCount and bail out
+		queueEntity["errorCount"] += 1
+		db.Put(queueEntity)
+		raise
+	# If that transportFunction did not raise an error that email has been successfully send
+	queueEntity["isSend"] = True
+	queueEntity["sendDate"] = utils.utcNow()
+	queueEntity["transportFuncResult"] = resultData
+	queueEntity.exclude_from_indexes.add("transportFuncResult")
+	db.Put(queueEntity)
+
+
+def sendEMail(*,
+			  tpl: str = None,
+			  stringTemplate: str = None,
+			  skel: Union[None, Dict, "SkeletonInstance", List["SkeletonInstance"]] = None,
+			  sender: str = None,
+			  dests: Union[str, List[str]] = None,
+			  cc: Union[str, List[str]] = None,
+			  bcc: Union[str, List[str]] = None,
+			  headers: Dict[str, str] = None,
+			  attachments: List[Tuple[str, str]] = None,
+			  **kwargs) -> Any:
+	"""
+	General purpose function for sending e-mail.
+	This function allows for sending e-mails, also with generated content using the Jinja2 template engine.
+	Your have to implement a method which should be called to send the prepared email finally. For this you have
+	to allocate *viur.email.transportFunction* in conf.
+
+	:param tpl: The name of a template from the deploy/emails directory.
+	:param stringTemplate: This string is interpreted as the template contents. Alternative to load from template file.
+		:param skel: The data made available to the template. In case of a Skeleton or SkelList, its parsed the usual way;\
+		Dictionaries are passed unchanged.
+	:param sender: The address sending this mail.
+	:param dests: A list of addresses to send this mail to. A bare string will be treated as a list with 1 address.
+	:param cc: Carbon-copy recipients. A bare string will be treated as a list with 1 address.
+	:param bcc: Blind carbon-copy recipients. A bare string will be treated as a list with 1 address.
+	:param headers: Specify headers for this email.
+	:param attachments: List of files ((filename, filecontent)-pairs) to be sent within the mail as attachments.
+
+	.. Warning:  As emails will be queued (and not send directly) you cannot exceed 1MB in total
```

Comment:
That's absolutely fine. You want this limit to be documented upfront (instead of being surprised in production later on). The datastore will check the size later on (no need to do this twice).